core/kernel/notif.c

```diff
@@ -106,6 +124,73 @@ void notif_send_async(uint32_t value)
 	cpu_spin_unlock_xrestore(&notif_lock, old_itr_status);
 }
 
+#if defined(CFG_CORE_IT_NOTIF)
+uint32_t notif_it_get_value(bool *it_valid, bool *it_pending)
+{
+	uint32_t old_itr_status = 0;
+	uint32_t res = 0;
+	int bit = 0;
+
+	old_itr_status = cpu_spin_lock_xsave(&it_lock);
+
+	bit_ffs(notif_it_pending, (int)NOTIF_IT_VALUE_MAX + 1, &bit);
+
+	*it_valid = (bit >= 0);
+	if (!*it_valid) {
+		*it_pending = false;
+		goto out;
+	}
+
+	res = bit;
+	bit_clear(notif_it_pending, res);
+	bit_ffs(notif_it_pending, (int)NOTIF_IT_VALUE_MAX + 1, &bit);
+	*it_pending = true;
```

Comment:
`s/true/(bit >= 0)/`

Reply:
At this stage, we know `bit >= 0` (lines 138..141).

Reply:
No, you have just cleared that bit. Now you're checking if there's any other bit to instruct the caller to call again to get the next bit.

Reply:
you're right! indeed. i'll fix, thanks.

Reply:
I forgot to fix this. I'll add a fixup commit.
My apologies.